src/fsharp/FSharp.Core/seq.fs

```diff
@@ -1488,3 +1488,88 @@ namespace Microsoft.FSharp.Collections
                                 [|SR.GetString SR.inputMustBePositive; count|]
             mkDelayedSeq (fun () ->
                 source |> toArray |> Array.splitInto count :> seq<_>)
+
+        [<CompiledName("RemoveAt")>]
+        let removeAt (index: int) (source: seq<'T>) : seq<'T> =
+            if index < 0 then invalidArg "index" "index must be within bounds of the array"
+            seq {
+                let mutable i = 0
+                for item in source do
+                    if i <> index then 
+                        yield item
+                    i <- i + 1
+                if i <= index then invalidArg "index" "index must be within bounds of the array"
+            }
+    
+        [<CompiledName("RemoveManyAt")>]
+        let removeManyAt (index: int) (count: int) (source: seq<'T>) : seq<'T> =
+            if index < 0 then invalidArg "index" "index must be within bounds of the array"
+            seq {
+                let mutable i = 0
+                for item in source do
+                    if i < index || i >= index + count then 
+                        yield item
+                    i <- i + 1
+                if i <= index then invalidArg "index" "index must be within bounds of the array"
+            }
+    
+        [<CompiledName("UpdateAt")>]
+        let updateAt (index: int) (value: 'T) (source: seq<'T>) : seq<'T> =
+            if index < 0 then invalidArg "index" "index must be within bounds of the array"
+            seq {
+                let mutable i = 0
+                for item in source do
+                    if i <> index then
+                        yield item
+                    else yield value
+                    i <- i + 1
+                if i <= index then invalidArg "index" "index must be within bounds of the array"
+            }
+    
+        [<CompiledName("UpdateManyAt")>]
+        let updateManyAt (index: int) (values: seq<'T>) (source: seq<'T>) : seq<'T> =
+            if index < 0 then invalidArg "index" "index must be within bounds of the array"
+            seq {
+                let mutable i = 0
+                let sourceEnumerator = source.GetEnumerator()
```

Comment:
This should be `use sourceEnumerator = source.GetEnumerator()`.  However I think we will be removing the ManyAt functions, I'll write a separate note about that.

Reply:
See https://github.com/fsharp/fslang-suggestions/issues/1047#issuecomment-894227703

Reply:
removed